### ISSUE TITLE
[1.0] Arrays are not supported as data property values

### ISSUE THREAD

**Issue by 0x80** [gatsbyjs/gatsby#873]
It looks like arrays as a property value in json data (as read by the source-filesystem plugin) are not supported at the moment. They give the same kind of funky error as a null value does.

```
UNHANDLED REJECTION Error: Can only create List of a GraphQLType but got: undefined.
    at invariant (/Users/me/Development/project/node_modules/graphql/jsutils/invariant.js:19:11)
    at new GraphQLList (/Users/me/Development/project/node_modules/graphql/type/definition.js:680:29)
    at exports.inferGraphQLInputFields (/Users/me/Development/project/node_modules/gatsby/dist/schema/infer-graphql-input-fields.js:91:25)
    at /Users/me/Development/project/node_modules/gatsby/dist/schema/infer-graphql-input-fields.js:146:25
```

**Comment by KyleAMathews** [comment]
Arrays are definitely supported. https://github.com/gatsbyjs/gatsby/blob/1.0/packages/gatsby/lib/schema/__tests__/infer-graphql-type-test.js#L16

It does look like something else is going on though. What is the data there that's being rejected?

Also, @danielfarrell and I were fixing things up with inferring the graphql schema yesterday which might have fixed your issue https://github.com/gatsbyjs/gatsby/pull/868

**Comment by 0x80** [comment]
With the latest 1.0 branch I still have this error. The data I'm trying to load is below. It's the toolTags property that Gatsby trips over.

```json
[{
  "title": "We Need Us",
  "slug": "we-need-us",
  "summary": "A sound engine for a data-driven artwork",
  "date": "2015",
  "toolTags": [{
    "name": "c/c++",
    "description": "C is a blabla"
  }]
}]
```

**Comment by KyleAMathews** [comment]
Hmmmm... yeah we're not testing arrays of objects like that. Could you add a test for it and figure out why it's failing? This is where to start investigating https://github.com/gatsbyjs/gatsby/blob/1.0/packages/gatsby/lib/schema/infer-graphql-type.js#L21

You could add an array of objects to the tests here: https://github.com/gatsbyjs/gatsby/blob/1.0/packages/gatsby/lib/schema/__tests__/infer-graphql-type-test.js

**Comment by 0x80** [comment]
@KyleAMathews I gave it a shot. Added some to the tests and tried to fix the recursion in `infer-graphql-type.js`, but I understand too little of graphql and the source to figure out what the solution should be. #881

**Comment by KyleAMathews** [comment]
Fixed now by #881
